src/Proxy/ContainerInterfaceProxy.php

```diff
@@ -0,0 +1,260 @@
+<?php
+
+namespace Yiisoft\Yii\Debug\Proxy;
+
+use Psr\Container\ContainerInterface;
+use Psr\Container\ContainerExceptionInterface;
+use Yiisoft\Di\ContainerProxyInterface;
+use Yiisoft\Proxy\ProxyManager;
+use Yiisoft\Yii\Debug\Event\ProxyMethodCallEvent;
+
+class ContainerInterfaceProxy implements ContainerProxyInterface
+{
+    public const LOG_ARGUMENTS = 1 << 0;
+
+    public const LOG_RESULT = 1 << 1;
+
+    public const LOG_ERROR = 1 << 2;
+
+    protected ContainerInterface $container;
+
+    private array $decoratedServices = [];
+
+    private ContainerProxyConfig $config;
+
+    private array $serviceProxy = [];
+
+    private ?object $currentError = null;
+
+    private ProxyManager $proxyManager;
+
+    public function __construct(
+        ContainerInterface $container,
+        ContainerProxyConfig $config
+    ) {
+        $this->config = $config;
+        $this->container = $container;
+        $this->proxyManager = new ProxyManager($this->config->getProxyCachePath());
+    }
+
+    public function withDecoratedServices(array $decoratedServices): ContainerProxyInterface
+    {
+        $proxy = clone $this;
+        $proxy->decoratedServices = array_merge($this->config->getDecoratedServices(), $decoratedServices);
+
+        return $proxy;
+    }
+
+    public function isActive(): bool
+    {
+        return $this->config->getIsActive() && $this->config->getDecoratedServices() !== [];
+    }
+
+    public function get($id, array $params = [])
+    {
+        $this->resetCurrentError();
+        try {
+            $instance = null;
+            $timeStart = microtime(true);
+            $instance = $this->getInstance($id, $params);
+        } catch (ContainerExceptionInterface $e) {
+            $this->repeatError($e);
+        } finally {
+            $this->log('get', [$id, $params], $instance, $timeStart);
+        }
+
+        if (is_object($instance) && $this->isDecorated($id) && (($proxy = $this->getServiceProxyCache($id)) || ($proxy = $this->getServiceProxy($id, $instance)))) {
+            $this->setServiceProxyCache($id, $proxy);
+            return $proxy;
+        }
+
+        return $instance;
+    }
+
+    public function has($id): bool
+    {
+        $this->resetCurrentError();
+        try {
+            $result = null;
+            $timeStart = microtime(true);
+            $result = $this->container->has($id);
+        } catch (ContainerExceptionInterface $e) {
+            $this->repeatError($e);
+        } finally {
+            $this->log('has', [$id], $result, $timeStart);
+            return $result;
+        }
+    }
+
+    protected function getCurrentResultStatus(): string
+    {
+        return $this->currentError === null ? 'success' : 'failed';
+    }
+
+    protected function repeatError(object $error): void
+    {
+        $this->currentError = $error;
+        $errorClass = get_class($error);
+        throw new $errorClass($error->getMessage());
```

Comment:
1. Why it needed?
2. You should do `throw $error` instead

Reply:
No,  I shouldn't, you're wrong, I tried that. Everything is right in this case.

Reply:
Not all errors supports the first parameter.
And by your case you are erase all previous errors

Reply:
Why did you resolve this?